src/sys/derivs.cxx

```diff
@@ -340,9 +340,9 @@ const Field3D D2DYDZ(const Field3D &f, CELL_LOC outloc, DIFF_METHOD method, REGI
       for(int k=0;k<f.getMesh()->LocalNz;k++) {
         int kp = (k+1) % (f.getMesh()->LocalNz);
         int km = (k-1+f.getMesh()->LocalNz) % (f.getMesh()->LocalNz);
-        result(i,j,k) = 0.25*( +(f(i,j+1,kp) - f(i,j-1,kp))/(coords->dy(i,j+1))
-                               -(f(i,j+1,km) - f(i,j-1,km))/(coords->dy(i,j-1)) )
-                    / coords->dz;
+        result(i,j,k) = 0.25*( +(f(i,j+1,kp) - f(i,j-1,kp))
+                               -(f(i,j+1,km) - f(i,j-1,km)) )
+                    / coords->dy(i,j) / coords->dz;
```

Comment:
Now assuming that dy is the same at j+1 and j-1

Reply:
Possibly I don't understand the original intention, but I don't think this assumption is being made: a non-uniform dy correction is not needed here because `dy` is constant in z (and `dz` is constant). The old version looks to me like DDY(f)[z+1] is uses dy(y+1) while DDY(f)[z-1] uses dy(y-1): I think they should both use dy(y) because dy is constant in z, and that this is what the new version is doing. When I test with https://github.com/boutproject/BOUT-dev/tree/test-difops, the old version does not converge, while the new one converges at 2nd order.

Reply:
Is it possible to merge ` / coords->dy(i,j) / coords->dz` into ` / (coords->dy(i,j) * coords->dz)` for clarity?